src/bootstrap/check.rs

```diff
@@ -301,6 +301,61 @@ impl Step for CodegenBackend {
     }
 }
 
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
+pub struct RustAnalyzer {
+    pub target: TargetSelection,
+}
+
+impl Step for RustAnalyzer {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.paths(&["src/tools/rust-analyzer", "rust-analyzer"])
```

Comment:
That second one is redundant, the builder already has logic that lets you abbreviate to the last component of the path.
```suggestion
        run.paths(&["src/tools/rust-analyzer"])
```